packages/puppeteer-core/src/common/Device.ts

```diff
@@ -14,23 +14,24 @@
  * limitations under the License.
  */
 
+import {Viewport} from './PuppeteerViewport.js';
+
 /**
  * @public
  */
 export interface Device {
-  name: string;
   userAgent: string;
-  viewport: {
-    width: number;
-    height: number;
-    deviceScaleFactor: number;
-    isMobile: boolean;
-    hasTouch: boolean;
-    isLandscape: boolean;
-  };
+  viewport: Viewport;
+}
+
+/**
+ * @public
+ */
+export interface KnownDevice extends Device {
+  name: string;
```

Comment:
Let's keep the name to be on the Device interface? Then there would be no need for a breaking change and a KnownDevice type.

Reply:
Done.